docs/design/editorial_design.md

```diff
@@ -0,0 +1,354 @@
+Designing an Edit API
+=====================
+
+# Design Overview
+The industry is full of timeline management software but nearly all of them include a fundamental set of tools for manipulation. The most well known being:
+
+- Overwrite
+- Trim
+- Cut
+- Slip
+- Slide
+- Ripple
+- Roll
+- Fill (3/4 Point Edit)
+
+OpenTimelineIO, while not seeking to become a fully realized NLE, could benefit from having a simple toolkit to help automate building/modifying timelines based on time rather than index alone.
+
+## Structure
+The underlying implementation for these edit commands will be in C++, possibly under a new library called `openedit`. This has the benefit of connecting with other languages more simply (Python, Swift, C, etc.) and, hopefully, we only need expose the functionality to the bindings.
+
+```
+OpenTimelineIO/
+    `- src/
+        `- opentime/
+        `- opentimelineio/
+        `- openedit/
+            `- ...
+```
+
+The benefit for the extra library is the edit suite will inevitably grow and will begin crowding the core model, which ultimately should be able to run without it.
+
+## Dependencies / Namespaces
+The `openedit` library will have to communicate with both `opentime` and `openedit`.
+
+This will warrant either a new namespace `namespace openedit { ... }` or using `opentimelineio` again. The latter makes some of the bellow comments less challenging but again, might pollute an otherwise clean namespace.
+
+### ErrorStatus
+This will present challenges when interfacing with the `ErrorStatus` of the different namespaces. That said, with a simple wrapper for this, we could have the instances auto convert as required.
+
+### Retainer
+Ideally, we continue to use the `Retainer<>` where possible to make sure editing is reference counted. Namespaces might make the names a bit clunky, seeing `SerializedObject::Retainer<>` all over:
+
+```
+namespace openedit { OPENEDIT_VERSION {
+
+using namespace opentimelineio::OPENTIMELINEIO_VERSION;
+auto item = SerializedObject::Retainer<Item>(...);
+
+} }
+```
+
+So perhaps establishing some common aliases early might help?
+```
+using RetainedItem = SerializedObject::Retainer<Item>;
+using RetainedComposition = SerializedObject::Retainer<Composition>;
+...
+```
+
+# Editing Commands
+Let's go through some of the most common commands and what the python API might look like:
+
+## Overwrite
+
+![Overwrite](../_static/edit/01_overwrite.png)
+- Anything overlapping is destroyed on contact. This may split an item.
```

Comment:
It would be nice to specify things more pedantically using the nomenclature we've established in the recent Allen's Interval Algebra commit. eg.

```
for all items i
   if e contains i, or i begins e, or e ends i : delete i
   if i contains e : split i at e.start, yielding {i0, i1}. Split i1 at en.end, yielding {i2, i3}, delete i, i1 and i2, return i0 and i3.
  if i overlaps e...
  else skip 
```

Reply:
You've specified the rest of the API more formally. I felt, given our new algebraic nomenclature, overlapping wasn't specific enough, also destroyed on contact was open to a bit of interpretation..

Reply:
Agreed - I've got ASCII art and more explicit instructions per-command on the way (taking in the additional notes where possible). I'll add the audience information to the top along with the replaced "elevator pitch" that @ssteinbach recommended which sums it up beautifully 👍